pkg/certs/certs.go

```diff
@@ -42,6 +44,75 @@ type LoadX509KeyPairFunc func(certFile, keyFile string) (tls.Certificate, error)
 // upon the TLS SNI sent by the client.
 type GetCertificateFunc func(hello *tls.ClientHelloInfo) (*tls.Certificate, error)
 
+// LoadCAs loads all X.509 certificates at the given path and adds them
+// to the list of system root CAs, if available. The returned CA pool
+// is a conjunction of the system root CAs and the certificate(s) at
+// the given path.
+//
+// If path is a regular file, LoadCAs simply adds it to the CA pool
+// if the file contains a valid X.509 certificate
+//
+// If the path points to a directory, LoadCAs iterates over all top-level
+// files within the directory and adds them to the CA pool if they contain
+// a valid X.509 certificate.
+func LoadCAs(path string) (rootCAs *x509.CertPool, err error) {
```

Comment:
how is this different from GetRootCAs() @aead - it is also designed to auto trust in k8s environments etc.

Reply:
Oh, yeah we can use `GetRootCAs` then - we just need to check if the path is a file and not just handle directories.